mobile/src/screens/BookScreen.tsx

```diff
@@ -0,0 +1,185 @@
+import { APIContext } from '../context/APIContext';
+import { I18nContext } from '../context/I18nContext';
+import { TextStyles } from '../styles/TextStyles';
+import { ButtonGroup } from '../components/ButtonGroup';
+import { LanguageButtons } from '../components/LanguageButtons';
+
+/**
+ * Individual book view displaying book details
+ */
+export const BookScreen: React.FC = () => {
+  // get book id and langs from route params
+  const route = useRoute<BookScreenRouteProps>();
+  const { id, langs } = route.params;
+
+  const client = useContext(APIContext);
+  const i18nCtx = useContext(I18nContext);
+
+  // book screen states
+  const [loading, setLoading] = useState(true);
+  const [language, setLanguage] = useState(i18nCtx.locale.substring(0, 2));
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [image, setImage] = useState('');
+  const [tabBody, setTabBody] = useState('');
+  const [tabVideo, setTabVideo] = useState('');
+  const [activeButton, setActiveButton] = useState('');
+  const [tabContent, setTabContent] = useState(null);
+
+  const markdownStyles = {
+    heading: { fontFamily: 'Gotham-Light' },
+    paragraph: { fontFamily: 'Gotham-Light' },
+    strong: { fontFamily: 'Gotham-Bold' },
+    listItemNumber: { fontFamily: 'Gotham-Light' },
+    listItemOrderedContent: { fontFamily: 'Gotham-Light' },
+    listItemUnorderedContent: { fontFamily: 'Gotham-Light' },
+    tableHeaderCellContent: { fontFamily: 'Gotham-Light' },
+    em: { fontFamily: 'Gotham-Italic' },
+  };
+
+  // fetches book details on language change
+  useEffect(
+    () => {
+      setLoading(true);
+      let lang;
+
+      if (langs.indexOf(language) > -1) {
+        lang = language;
+      } else if (langs.indexOf('en') > -1) {
+        lang = 'en';
+        setLanguage(lang);
+      } else {
+        const [first] = langs;
+        lang = first;
+        setLanguage(lang);
+      }
+
+      client.getBook(id, lang).then((res) => {
+        setTitle(res.title);
+        setAuthor(res.author);
+        setImage(res.image);
+        setActiveButton('btn-1');
+        setTabContent({
+          'btn-1': res.read,
+          'btn-2': res.explore,
+          'btn-3': res.learn,
+        });
+        setLoading(false);
+      }).catch((err) => {
+        console.log(err);
+      });
+    },
+    [language],
+  );
+
+  // sets tab content on tab button change
+  useEffect(
+    () => {
+      if (tabContent) {
+        setTabVideo(null);
+        if (tabContent[activeButton].video) {
+          const regEx = '^(?:https?:)?//[^/]*(?:youtube(?:-nocookie)?.com|youtu.be).*[=/]([-\\w]{11})(?:\\?|=|&|$)';
+          const matches = tabContent[activeButton].video.match(regEx);
+          if (matches) {
+            setTabVideo(matches[1]);
+          }
+        }
+        setTabBody(tabContent[activeButton].body);
+      }
+    },
+    [activeButton, tabContent],
+  );
```

Comment:
I think you should have a state var for the selected tab of type `'read' | 'explore' | 'learn'`, then you won't need an effect. You can just pass `book[selectedTab]` into the youtube and markdown renderers